### ISSUE TITLE
Series: non-zero padded

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#382]
**Describe the bug**
It seems that with the latest updates we cannot open `fileBased` series without zero-padding anymore, e.g.:
```
simData_0.h5
simData_500.h5
simData_1500.h5
simData_2000.h5
```

**To Reproduce**

Python:
```python
import openPMD

s = openPMD.Series('path/to/simData_%T.h5', openPMD.Access_Type.read_only)
# RuntimeError: Can not determine iteration padding from existing filenames.
# Please specify '%0<N>T'.

s = openPMD.Series('fields_%04T.h5', openPMD.Access_Type.read_only)
# only subset (ok, but somehow being able to select all should work, too)
list(s.iterations)
# [1500, 2000]
```
or C++
```C++
#include <openPMD/openPMD.hpp>

auto s = openPMD::Series('path/to/simData_%T.h5', AccessType::READ_ONLY)
```

**Expected behavior**
Should we be able to reads those as well? Did I mix up the syntax for it?

**Software Environment:**
 - version of openPMD-api: 0.6.2-alpha
 - installed openPMD-api via: pip
 - operating system: Ubuntu 14.04.5 LTS "trusty"
 - machine: Hypnos5 (Jupyter Notebook queue)
 - name and version of Python implementation:CPython 3.6.2
 - version of HDF5: 1.8.14 (serial)
 - version of ADIOS1: *none*
 - name and version of MPI: *none*

**Comment by ax3l** [comment]
cc @C0nsultant am I doing something wrong here?

**Comment by ax3l** [comment]
Proposed a fix in #388.
